api/src/main/java/com/datastrato/gravitino/authorization/Resources.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import java.util.Objects;
+
+/** The helper class for {@link Resource}. */
+public class Resources {
+
+  /**
+   * Create the {@link Resource} with the given names.
+   *
+   * @param names The names of the resource
+   * @return The created {@link Resource}
+   */
+  public static Resource of(String... names) {
```

Comment:
How do users use this `Resource` concept, what is the relationship to the catalogs, schemas, and tables? I think the introduction of `Resource` increase the understanding burden to users, users may not know how to use it, I was thinking if there's a better way to clearly define it, which will be easy for users to use.

Reply:
How about letting me user pass the names like
`ofCatalog`. `ofCatalog.ofSchema`.

Reply:
Databricks has the similar concept `securable objects `.
Snowflake use `securable object`, too. I will change the name to `securable object`.

Reply:
Done.